diff --git a/CMakeLists.txt b/CMakeLists.txt
index afa0461..b58faad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,23 +93,14 @@ endif()
 
 #########################################################
 # Static library (as base for other targets)
-add_library(${PROJECT_NAME}_s STATIC ${SRCS_MAIN_PLATFORM} ${SRCS_COMMON})
-target_include_directories(${PROJECT_NAME}_s PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/Include)
-target_include_directories(${PROJECT_NAME}_s PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/Include)
-SET_TARGET_PROPERTIES(${PROJECT_NAME}_s PROPERTIES PREFIX "" IMPORT_PREFIX "")
+add_library(${PROJECT_NAME} STATIC ${SRCS_MAIN_PLATFORM} ${SRCS_COMMON})
+target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/Include)
+target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/Include)
+SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES PREFIX "" IMPORT_PREFIX "")
 if (UNIX OR MINGW)
-	target_compile_options(${PROJECT_NAME}_s PUBLIC "-fPIC")
-	# Uncomment the following to but the version info into the .so-file.
-	SET_TARGET_PROPERTIES(${PROJECT_NAME}_s PROPERTIES VERSION "${LIB3MF_VERSION_MAJOR}.${LIB3MF_VERSION_MINOR}.${LIB3MF_VERSION_MICRO}.${BUILD_NUMBER}")
-	SET_TARGET_PROPERTIES(${PROJECT_NAME}_s PROPERTIES SOVERSION "${LIB3MF_VERSION_MAJOR}")
-	if (NOT APPLE)
-		SET_TARGET_PROPERTIES(${PROJECT_NAME}_s PROPERTIES LINK_FLAGS -s)
-	endif()
-	find_library(LIBUUID_PATH uuid)
-	if(NOT LIBUUID_PATH)
-		message(FATAL_ERROR "libuuid not found")
-	endif()
-	target_link_libraries(${PROJECT_NAME}_s ${LIBUUID_PATH})
+	target_compile_options(${PROJECT_NAME} PUBLIC "-fPIC")
+	SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES VERSION "${LIB3MF_VERSION_MAJOR}.${LIB3MF_VERSION_MINOR}.${LIB3MF_VERSION_MICRO}.${BUILD_NUMBER}")
+	target_link_libraries(${PROJECT_NAME} ${LIBUUID_PATH})
 else()
 	# wd4996 masks the deprecated-warning
 	target_compile_options(${PROJECT_NAME}_s PUBLIC "$<$<CONFIG:DEBUG>:/Od;/Ob0;/Gm;/sdl;/W3;/WX;/FC;/wd4996>")
@@ -119,13 +110,8 @@ endif()
 
 #########################################################
 # Shared library
-add_library(${PROJECT_NAME} SHARED ${SRCS_MAIN_PLATFORM}
-	"./Source/Model/COM/NMR_DLLInterfaces.cpp"
-	${VERSION_FILES_OUTPUTLOCATION}
-)
 set(STARTUPPROJECT ${PROJECT_NAME})
-SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES PREFIX "" IMPORT_PREFIX "")
-target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}_s)
+target_link_libraries(${PROJECT_NAME})
 if (UNIX OR MINGW)
 	# Uncomment the following to but the version info into the .so-file.
 	SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES VERSION "${LIB3MF_VERSION_MAJOR}.${LIB3MF_VERSION_MINOR}.${LIB3MF_VERSION_MICRO}.${BUILD_NUMBER}")
